pkg/yurthub/multiplexer/manager.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multiplexer
+
+import (
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	kmeta "github.com/openyurtio/openyurt/pkg/yurthub/kubernetes/meta"
+	ystorage "github.com/openyurtio/openyurt/pkg/yurthub/multiplexer/storage"
+)
+
+var keyFunc = func(obj runtime.Object) (string, error) {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return "", err
+	}
+
+	name := accessor.GetName()
+	if len(name) == 0 {
+		return "", apierrors.NewBadRequest("Name parameter required.")
+	}
+
+	ns := accessor.GetNamespace()
+	if len(ns) == 0 {
+		return "/" + name, nil
+	}
+	return "/" + ns + "/" + name, nil
+}
+
+var attrsFunc = func(obj runtime.Object) (labels.Set, fields.Set, error) {
+	metadata, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var fieldSet fields.Set
+	if len(metadata.GetNamespace()) > 0 {
+		fieldSet = fields.Set{
+			"metadata.name":      metadata.GetName(),
+			"metadata.namespace": metadata.GetNamespace(),
+		}
+	} else {
+		fieldSet = fields.Set{
+			"metadata.name": metadata.GetName(),
+		}
+	}
+
+	return labels.Set(metadata.GetLabels()), fieldSet, nil
+}
+
+type MultiplexerManager interface {
+	ResourceCacheConfig(gvr *schema.GroupVersionResource) (*ResourceCacheConfig, error)
+	ResourceCache(gvr *schema.GroupVersionResource) (Interface, func(), error)
+}
+
+type multiplexerManager struct {
+	restStoreManager    ystorage.StorageManager
+	restMapper          meta.RESTMapper
+	cacheMap            map[string]Interface
+	cacheConfigMap      map[string]*ResourceCacheConfig
+	cacheDestroyFuncMap map[string]func()
+}
+
+func NewRequestsMultiplexerManager(
+	restStoreManager ystorage.StorageManager) MultiplexerManager {
+
+	return &multiplexerManager{
+		restStoreManager:    restStoreManager,
+		restMapper:          kmeta.NewDefaultRESTMapperFromScheme(),
```

Comment:
The RestScope in `kmeta.NewDefaultRESTMapperFromScheme()` is fixed as `namespace` as following, I think we need to improve `kmeta.NewDefaultRESTMapperFromScheme()` for multiplexer.

https://github.com/openyurtio/openyurt/blob/09b1ab1db34c0f9d7e70219ea72acdb281341c72/pkg/yurthub/kubernetes/meta/restmapper.go#L75

Reply:
namespaced scope is not used.